numba/tests/test_comprehension.py

```diff
@@ -251,6 +225,166 @@ def list24(x):
             msg = "Invalid usage of == with parameters"
             self.assertIn(msg, str(raises.exception))
 
+class TestArrayComprehension(unittest.TestCase):
+
+    @tag('important')
+    def test_comp_with_array_1(self):
```

Comment:
I've made a PR with the dtype refinement on your branch.  https://github.com/IntelLabs/numba/pull/18